### PR TITLE
Permissions

### PR DESCRIPTION
**Pull request by ptone**
this branch is for development of permission checks for package and grid modification - work in progress

### CODE REVIEW

**Comment by ptone** [comment]
a single pair of tests are added for adding package.

remaining tests are needed to cover the other permissions and views

File: apps/package/tests/test_views.py [hunk 1]
```diff
@@ -66,6 +70,26 @@ class FunctionalPackageTest(TestCase):
         self.assertEqual(response.status_code, 302)
         self.assertEqual(Package.objects.count(), count + 1)
 
+    def test_add_package_permission_fail(self):
+        url = reverse('add_package')
+        self.assertTrue(self.client.login(username='user', password='user'))
+        settings.RESTRICT_PACKAGE_EDITORS = True
```
Comment (pydanny) [inline apps/package/tests/test_views.py:76]: See how this sets the permission! to TRUE

File: apps/package/tests/test_views.py [hunk 2]
```diff
@@ -66,6 +70,26 @@ class FunctionalPackageTest(TestCase):
...
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 403)
+        settings.RESTRICT_PACKAGE_EDITORS = False
+
+    def test_add_package_permission_success(self):
+        url = reverse('add_package')
+        self.assertTrue(self.client.login(username='user', password='user'))
+        settings.RESTRICT_PACKAGE_EDITORS = True
+        user = User.objects.get(username='user')
+        add_package_perm = Permission.objects.get(codename="add_package")
```
Comment (pydanny) [inline apps/package/tests/test_views.py:86]: Tricky party: Retrieve the permission based on this code name ('add_<model-name>')

File: apps/package/tests/test_views.py [hunk 3]
```diff
@@ -66,6 +70,26 @@ class FunctionalPackageTest(TestCase):
...
+        user.user_permissions.add(add_package_perm)
```
Comment (pydanny) [inline apps/package/tests/test_views.py:87]: Hand it to the permissions add group.

**Pull request closed without merging**